src/pages/new-chat/new-chat.ts

```diff
@@ -41,28 +52,59 @@ export class NewChatComponent implements OnInit {
     });
   }
 
-  private findUsers(): ObservableCursor<User> {
+  findContacts(): Promise<Contact[]> {
+    // If we're running this in the browser, don't look for any contacts
+    if (!navigator.hasOwnProperty('contacts')) return Promise.resolve();
+
+    const fields = <ContactFieldType[]>['phoneNumbers'];
+
+    // Look for all the available phone numbers in contacts
+    return Contacts.find(fields, {
+      hasPhoneNumber: true,
+      multiple: true,
+      desiredFields: fields
+    });
+  }
+
+  subscribeUsers(contacts?: any[]): Subscription {
+    // Fetch all users in contacts list. If no contacts list was provided, will
+    // fetch all available users in database. This behavior is **not** recommended
+    // in a production application since it will probably overload both client
+    // and server
+    return MeteorObservable.subscribe('users', contacts).subscribe(() => {
+      MeteorObservable.autorun().subscribe(() => {
+        this.users = this.findUsers().zone();
+      });
+    });
+  }
+
+  findUsers(): ObservableCursor<User> {
+    // Find all belonging chats
     return Chats.find({
       memberIds: this.senderId
     }, {
       fields: {
         memberIds: 1
       }
     })
-      .startWith([]) // empty result
-      .mergeMap((chats) => {
-        const recieverIds = chats
-          .map(({memberIds}) => memberIds)
-          .reduce((result, memberIds) => result.concat(memberIds), [])
-          .concat(this.senderId);
+    // The initial value of the upcoming mapping function
+    .startWith([])
+    .mergeMap((chats) => {
+      // Get all userIDs with whom we're chatting with
+      const recieverIds = _.chain(chats)
+        .pluck('memberIds')
+        .flatten()
+        .concat(this.senderId)
+        .value();
 
-        return Users.find({
-          _id: {$nin: recieverIds}
-        })
-      });
+      // Find all users which are not in belonging chats
+      return Users.find({
+        _id: { $nin: recieverIds }
+      })
+    });
   }
 
-  private handleError(e: Error): void {
+  handleError(e: Error): void {
```

Comment:
I think that the error object in this case of `Meteor.Error` and not just `Error` of JavaScript, because the error comes from the `MeteorObservable.call` error callback...